code/game/area/areas.dm

```diff
@@ -548,3 +555,131 @@ GLOBAL_LIST_EMPTY(teleportlocs)
 	if(name == initial(name))
 		return name
 	return "[name] ([initial(name)])"
+
+// Day night stuff
+/**
+ * This will calculate all adjacent turfs in this area and add them to a cache for lighting effects.
+ *
+ * WARNING: This proc is VERY expensive and should be used sparingly.
+ */
+/area/proc/initialize_day_night_adjacent_turfs()
+	LAZYCLEARLIST(adjacent_day_night_turf_cache)
+	LAZYINITLIST(adjacent_day_night_turf_cache)
+
+	for(var/turf/iterating_turf in contents)
```

Comment:
![image](https://user-images.githubusercontent.com/75460809/233891522-e9387ba0-2a04-4a85-9d56-aedd930261f7.png)

Reply:
Please do not use code review for unintelligible feedback